### DISCUSSION TITLE
http.createServer is not a function

### DISCUSSION THREAD

**Discussion by thevalo** [socketio/socket.io#4642] [Q&A] [answered]
Hi, I'm actually abusing a few of my last brain cells trying to setup socket.io on my Inertia, Vite and Vue app. 
First of all, I'm not sure if it's solely the problem with socket.io but I'm not sure whether to ask for help elsewhere.

I've seen some solutions on the internet, but can't really find an appropriate answer for this problem.

Code:
```
import { Server } from "socket.io";
import { io } from "socket.io-client";


const socket = io();

socket.on("event", () => {
    console.log("got an event") ;
});


const server = new Server();

server.on("connection", (socket) => {
    console.log('server connect...');
});

server.listen(3000);
```

The error:
`Uncaught (in promise) TypeError: http.createServer is not a function`
Full screenshot from console
![image](https://user-images.githubusercontent.com/28608203/221578887-7e9a080e-9982-4d6b-a691-0a7207228ee5.png)

What's the problem here? I see that the problem is on the server side, and is happening right when it's creating a http server and binding but I'm perplexed whether I've set it up correctly or not?

Using versions:

 "socket.io": "^4.6.1",
 "socket.io-client": "^4.6.1",

"vue": "^3.2.47",
"@inertiajs/vue3": "^1.0.1",

**Comment by thevalo** [comment]
Seems like installing and importing express like this  `import express from 'express';`, fixed the initial problem, now this error is gone but another one appeared:

`Uncaught (in promise) TypeError: Cannot read properties of null (reading 'ReadStream')`

Screenshot:
![image](https://user-images.githubusercontent.com/28608203/221795541-2f8933e9-04ea-484d-af9b-32362f3dafb1.png)

Which led me think that this is not related to socket.io?

**Comment by AndiLeni** [comment, marked as answer]
Did you try to run this code in the browser?
A server can only be created from the nodejs backend.
